x-pack/legacy/plugins/maps/public/elasticsearch_geo_utils.js

```diff
@@ -75,17 +75,17 @@ export function hitsToGeoJson(hits, flattenHit, geoFieldName, geoFieldType) {
     // don't include geometry field value in properties
     delete properties[geoFieldName];
 
-    // _id is unique to Elasticsearch documents.
-    // Move _id to FEATURE_ID_PROPERTY_NAME to standardize featureId keys across all sources
-    properties[FEATURE_ID_PROPERTY_NAME] = properties._id;
-    delete properties._id;
-
     //create new geojson Feature for every individual geojson geometry.
     for (let j = 0; j < tmpGeometriesAccumulator.length; j++) {
       features.push({
         type: 'Feature',
         geometry: tmpGeometriesAccumulator[j],
-        properties: properties
+        properties: {
```

Comment:
This recreates the `properties`-object, copying everything. the reason the previous code specifically _did not_ do this (https://github.com/elastic/kibana/pull/48594/files#diff-9c78f255b553d9556966390c977b5a08L80-L81) is because decoding should execute as tightly as possible. These are tight loops that benefit from being fast and have low mem-footprint. The main bottleneck is geometries (which can have deeply nested coordinate-arrays), but the same principle applies for properties. There's no harm in updating `properties` in place and copying the reference because the geojson collection is not reused elsewhere.

Reply:
As discussed offline, there is no way around this because otherwise we need a unique id per geometry in a document. I created issue https://github.com/elastic/kibana/issues/48829 to track how we handle array's of geometries.

The performance issues are much smaller then deep cloning the geometry since its just a single object where the geometry could contain multitudes of nested arrays